python/metric_pipeline_tasks/TractMeasureTasks.py

```diff
@@ -0,0 +1,131 @@
+import astropy.units as u
+from astropy.table import join
+import numpy as np
+from lsst.pipe.base import Struct, Task
+from lsst.pex.config import Config
+from lsst.verify import Measurement, Datum
+from lsst.pex.config import Config, Field, ListField
+from metric_pipeline_utils.stellar_locus import stellarLocusResid, calcQuartileClippedStats
+from lsst.sims.catUtils.dust.EBV import EBVbase as ebv
+
+
+class WPerpTaskConfig(Config):
+    # These are cuts to apply to the r-band only:
+    bright_rmag_cut = Field(doc="Bright limit of catalog entries to include",
+                            dtype=float, default=17.0)
+    faint_rmag_cut = Field(doc="Faint limit of catalog entries to include",
+                           dtype=float, default=23.0)
+
+
+class WPerpTask(Task):
+    ConfigClass = WPerpTaskConfig
+    _DefaultName = "WPerpTask"
+
+    def run(self, catalogs, photo_calibs, metric_name, vIds):
+        self.log.info(f"Measuring {metric_name}")
+        patches = set([p['patch'] for p in vIds])
+        bands = set([f['abstract_filter'] for f in vIds])
+        # bands = set([f['band'] for f in vIds])
+
+        # Extinction coefficients for HSC filters for conversion from E(B-V) to extinction, A_filter.                                                                             
+        # Numbers provided by Masayuki Tanaka (NAOJ).                                                                                                                             
+        #                                                                                                                                                                         
+        # Band, A_filter/E(B-V)                                                                                                                                                   
+        extinctionCoeffs_HSC = {
+            "g": 3.240,
+            "r": 2.276,
+            "i": 1.633,
+            "z": 1.263,
+            "y": 1.075,
+            "HSC-G": 3.240,
+            "HSC-R": 2.276,
+            "HSC-I": 1.633,
+            "HSC-Z": 1.263,
+            "HSC-Y": 1.075,
+            "NB0387": 4.007,
+            "NB0816": 1.458,
+            "NB0921": 1.187,
+        }
+
+        if ('g' in bands) & ('r' in bands) & ('i' in bands):
```

Comment:
From a coding style perspective, I feel like there is some code duplication here that could be avoided, for example, by creating a dictionary to hold the catalogs for the (g, r, i) bands, as well as dictionary for the magnitudes and magnitude errors. The bands could be used as keys.

Reply:
Hopefully the refactoring I did addresses this, but suggestions for smoother refactoring are welcome.